internal/pkg/admission/policy-server-deployment_test.go

```diff
@@ -432,3 +432,118 @@ func TestDefaultContainerSecurityContext(t *testing.T) {
+		}
+		expectedEnvVarPosition := 2
+		if envIndex != expectedEnvVarPosition {
+			t.Error("Function must find a metrics environment at position {}. Found at {}.", expectedEnvVarPosition, envIndex)
+		}
+
+		env = []corev1.EnvVar{{Name: "env1"}, {Name: "env2"}, {Name: "env3"}}
+		container = corev1.Container{
+			Image: "image",
+			Env:   env,
+			Name:  policyServerContainerName,
+		}
+		hasEnvvar, envIndex = envVarsContaintsVariable(&container, envVarName)
+		if hasEnvvar {
+			t.Error("Function must not find a metrics environment variable.")
+		}
+		expectedEnvVarPosition = -1
+		if envIndex != expectedEnvVarPosition {
+			t.Error("Function must the metrics environment variable at position {}. Found at {}.", expectedEnvVarPosition, envIndex)
+		}
+	}
+}
+
+func TestPolicyServerDeploymentMetricConfigurationWithValueDefinedByUser(t *testing.T) {
+	reconciler := Reconciler{
+		Client:               nil,
+		DeploymentsNamespace: "kubewarden",
+		MetricsEnabled:       true,
+	}
+	policyServer := &policiesv1.PolicyServer{
+		Spec: policiesv1.PolicyServerSpec{
+			Image: "image",
+			Env:   []corev1.EnvVar{{Name: "KUBEWARDEN_ENABLE_METRICS", Value: "0"}, {Name: "KUBEWARDEN_LOG_FMT", Value: "invalid"}},
+		},
+	}
+	deployment := reconciler.deployment("v1", policyServer)
+	hasMetricEnvvar := false
+	hasLogFmtEnvvar := false
+	for _, envvar := range deployment.Spec.Template.Spec.Containers[0].Env {
+		if envvar.Name == "KUBEWARDEN_ENABLE_METRICS" {
+			hasMetricEnvvar = true
+			if envvar.Value != "1" {
+				t.Error("Missing KUBEWARDEN_ENABLE_METRICS value")
+			}
+		}
+		if envvar.Name == "KUBEWARDEN_LOG_FMT" {
+			hasLogFmtEnvvar = true
+			if envvar.Value != "otlp" {
+				t.Error("Missing KUBEWARDEN_LOG_FMT value")
+			}
+		}
+	}
+	if !hasMetricEnvvar {
+		t.Error("Missing KUBEWARDEN_ENABLE_METRICS environment variable")
+	}
+	if !hasLogFmtEnvvar {
+		t.Error("Missing KUBEWARDEN_LOG_FMT environment variable")
+	}
+
+	value, hasAnnotation := deployment.Annotations["sidecar.opentelemetry.io/inject"]
+	if !hasAnnotation {
+		t.Error("Missing OPTEL annotation")
+	}
+	if value != "true" {
+		t.Error("OPTEL annotation invalid value")
+	}
+}
+
+func TestPolicyServerDeploymentMetricConfigurationWithNoValueDefinedByUSer(t *testing.T) {
+	reconciler := Reconciler{
+		Client:               nil,
+		DeploymentsNamespace: "kubewarden",
+		MetricsEnabled:       false,
+	}
+	policyServer := &policiesv1.PolicyServer{
+		Spec: policiesv1.PolicyServerSpec{
+			Image: "image",
+			Env:   []corev1.EnvVar{},
+		},
+	}
+	deployment := reconciler.deployment("v1", policyServer)
+	hasMetricEnvvar := false
+	hasLogFmtEnvvar := false
+	for _, envvar := range deployment.Spec.Template.Spec.Containers[0].Env {
+		if envvar.Name == "KUBEWARDEN_ENABLE_METRICS" {
+			hasMetricEnvvar = true
+		}
+		if envvar.Name == "KUBEWARDEN_LOG_FMT" {
+			hasLogFmtEnvvar = true
+		}
+	}
+	if hasMetricEnvvar {
+		t.Error("KUBEWARDEN_ENABLE_METRICS should not define environment variable")
```

Comment:
```suggestion
		t.Error("KUBEWARDEN_ENABLE_METRICS should not be set")
```

(same for the other messages)